Common/Tests/Utilities.UI/UI/ReplWindowProxy.cs

```diff
@@ -349,6 +411,22 @@ string executionMode
                 .Select(l => l.TrimEnd('\r', '\n', ' '))
                 .ToList();
 
+#if DEV14_OR_LATER
+            dynamic eval = _window.Evaluator;
+            var primary = eval.PrimaryPrompt as string ?? ">>>";
+#else
```

Comment:
Files under `Common` should keep all ifdefs, files under `Python` should keep back to DEV14 but should not add new ones for DEV14 (this is to help people backport if they need to).

Python 2.5 support is nearly dropped, due to our dependency on json for so many features, but in general I think we need to preemptively expand our portability for C and Python code (versions/platforms/interpreters/etc.) while not worrying about it for C# code until we need it.